tests/31sync/06state.pl

```diff
@@ -338,6 +338,64 @@
    };
 
 
+test "A change to displayname should not result in a full state sync",
+   requires => [qw( first_api_client can_sync )],
+   bug => 'SYN-515',
+
+   check => sub {
+      my ( $http ) = @_;
+
+      my ( $user, $filter_id, $room_id, $next_batch );
+
+      my $filter = {
+         room => {
+            timeline  => { types => [] },
+            state     => { types => [ "a.madeup.test.state" ] },
+            ephemeral => { types => [] },
+         },
+         presence => {types => [] },
+      };
+
+      matrix_register_user_with_filter( $http, $filter )->then( sub {
+         ( $user, $filter_id ) = @_;
+
+         matrix_create_room( $user );
+      })->then( sub {
+         ( $room_id ) = @_;
+
+         matrix_put_room_state( $user, $room_id,
+            type      => "a.madeup.test.state",
+            content   => { "my_key" => 1 },
+            state_key => ""
+         );
+      })->then( sub {
+         matrix_sync( $user, filter => $filter_id );
+      })->then( sub {
+         my ( $body ) = @_;
+
+         $next_batch = $body->{next_batch};
+         @{ $body->{rooms}{joined}{$room_id}{state}{events} } == 1
+            or die "Expected one state event";
+
+         matrix_put_room_state( $user, $room_id,
+            type      => "m.room.member",
+            content   => { "membership" => "join",
+                           "displayname" => "boris" },
+            state_key => $user->user_id,
+         );
+      })->then( sub {
+         matrix_sync( $user, filter => $filter_id );
+      })->then( sub {
+         my ( $body ) = @_;
+
+         @{ $body->{rooms}{joined}{$room_id}{state}{events} } == 0
```

Comment:
Aren't we expecting to see the display change state event? Or does this PR assume that the state change will only be in the timeline?